Espressif/ESP32/ESP32-SSH-Server/main/main.c

```diff
@@ -242,42 +130,65 @@ bool NoEthernet()
 #if defined(WOLFSSH_SERVER_IS_AP) || defined(WOLFSSH_SERVER_IS_STA)
 void init_nvsflash()
 {
+    esp_err_t ret = ESP_OK;
     ESP_LOGI(TAG, "Setting up nvs flash for WiFi.");
-    esp_err_t ret = nvs_flash_init();
+
+    ret = nvs_flash_init();
+
+#if defined(CONFIG_IDF_TARGET_ESP8266)
+    if (ret == ESP_ERR_NVS_NO_FREE_PAGES) {
+        ESP_ERROR_CHECK(nvs_flash_erase());
+        ret = nvs_flash_init();
+    }
+#else
     if (ret == ESP_ERR_NVS_NO_FREE_PAGES
           ||
-        ret == ESP_ERR_NVS_NEW_VERSION_FOUND) {
+        ret == ESP_ERR_NVS_NEW_VERSION_FOUND
+       ) {
 
         ESP_ERROR_CHECK(nvs_flash_erase());
         ret = nvs_flash_init();
     }
+#endif
+
     ESP_ERROR_CHECK(ret);
 }
 #endif
 
 /*
  * main initialization for UART, optional ethernet, time, etc.
  */
-void init()
+int init(void)
 {
+    int ret = ESP_OK;
     TickType_t EthernetWaitDelayTicks = (1000 / portTICK_PERIOD_MS);
 
     ESP_LOGI(TAG, "Begin main init.");
 
-#ifdef DEBUG_WOLFSSH
-    ESP_LOGI(TAG, "wolfSSH debugging on.");
-    wolfSSH_Debugging_ON();
-#endif
-
+    #ifdef DEBUG_WOLFSSH
+    {
+        ESP_LOGI(TAG, "wolfSSH debugging on.");
+        wolfSSH_Debugging_ON();
+    }
+    #endif
 
-#ifdef DEBUG_WOLFSSL
-    ESP_LOGI(TAG, "wolfSSL debugging on.");
-    wolfSSL_Debugging_ON();
-    ESP_LOGI(TAG,"Debug ON");
+    #ifdef DEBUG_WOLFSSL
+    {
+        ESP_LOGI(TAG, "wolfSSL debugging on.");
+        wolfSSL_Debugging_ON();
+        ESP_LOGI(TAG, "Debug ON");
+    }
     /* TODO ShowCiphers(); */
-#endif
+    #endif
 
+    /* Set time for cert validation.
+     * Some lwIP APIs, including SNTP functions, are not thread safe. */
+    ret = set_time(); /* need to setup NTP before WiFi */
+
+#ifndef DISABLE_SSH_UART
```

Comment:
Is it possible to just send message to SSH client when disabled UART instead to put messages to UART? This should be nice if user doesn't have UART connection.

Reply:
@miyazakh I can certainly make the default to be no wolfSSH / wolfSSL debugging messages. I had purposely left it on as it seemed more interesting for an example. 

Would you prefer I also turn off other verbosity, such as when characters are sent / received? For example, I've disabled all wolfSSL and wolfSSH debugging, yet these messages still appear in the serial port at the server:

```text
I (99788) ssh_server: Received 1 bytes from client.
I (99795) uart_helper: UART Send Data
I (99795) TX_TASK: Wrote 1 bytes
I (124278) SSH Server main: wolfSSH Server main loop heartbeat!
```

There's also a variety of other initialization not controlled by the wolfSSL gating macros. I could make it more silent with instructions for verbosity in the `README.md`. 

Let me know which you prefer.

Reply:
That's an interesting request. The whole point of the example is the SSH to UART, but let me see if I can do something for only the SSH server part with no connected UART.

How did you envision this working? Perhaps just echo the characters to the server console USB connected UART, like a debug message, noted [above](https://github.com/wolfSSL/wolfssh-examples/pull/4#discussion_r1479125590) in debugging?

Reply:
I mean that server sends message to the ssh client terminal, like welcome message does.
![Teraterm](https://github.com/wolfSSL/wolfssh-examples/assets/29424455/a46fa84b-792c-4d94-9776-a00f63f8fb8c)

Reply:
It appears that setting `SSH_SERVER_ECHO` could achieves the purpose.